app/models/social_networking/profile_answer.rb

```diff
@@ -10,5 +10,8 @@ class ProfileAnswer < ActiveRecord::Base
                foreign_key: :social_networking_profile_question_id
 
     validates :profile, :profile_question, presence: true
+    validates :profile_question, uniqueness: {
```

Comment:
It will be more efficient if you validate the foreign key uniqueness instead:

```ruby
validates :profile_question_id, uniqueness: {
  scope: :profile_id, message: "has already been answered"
}
```

This way Rails won't have to load the ProfileQuestion and Profile models.